internal/saucecloud/xcuit.go

```diff
@@ -0,0 +1,107 @@
+package saucecloud
+
+import (
+	"fmt"
+
+	"github.com/rs/zerolog/log"
+
+	"github.com/saucelabs/saucectl/internal/job"
+	"github.com/saucelabs/saucectl/internal/xcuit"
+)
+
+// XcuitRunner represents the Sauce Labs cloud implementation for xcuit.
+type XcuitRunner struct {
+	CloudRunner
+	Project xcuit.Project
+}
+
+// RunProject runs the tests defined in xcuit.Project.
+func (r *XcuitRunner) RunProject() (int, error) {
+	exitCode := 1
+
+	if err := r.validateTunnel(r.Project.Sauce.Tunnel.ID); err != nil {
+		return exitCode, err
+	}
+
+	appFileID, err := r.uploadProject(r.Project.Xcuit.App, appUpload)
+	if err != nil {
+		return exitCode, err
+	}
+
+	testAppFileID, err := r.uploadProject(r.Project.Xcuit.TestApp, testAppUpload)
+	if err != nil {
+		return exitCode, err
+	}
+
+	passed := r.runSuites(appFileID, testAppFileID)
+	if passed {
+		exitCode = 0
+	}
+
+	return exitCode, nil
+}
+
+func (r *XcuitRunner) runSuites(appFileID, testAppFileID string) bool {
+	sigChan := r.registerSkipSuitesOnSignal()
+	defer unregisterSignalCapture(sigChan)
+
+	jobOpts, results, err := r.createWorkerPool(r.Project.Sauce.Concurrency)
+	if err != nil {
+		return false
+	}
+	defer close(results)
+
+	// Submit suites to work on.
+	jobsCount := r.calculateJobsCount(r.Project.Suites)
+	go func() {
+		for _, s := range r.Project.Suites {
+			for _, d := range s.Devices {
+				log.Debug().Str("suite", s.Name).Str("device", d.Name).Str("platformVersion", d.PlatformVersion).Msg("Starting job")
```

Comment:
Should we also display deviceId ? All info will be blank when a specific device is requested.

Reply:
I added deviceID to the logs.